variables.tf

```diff
@@ -23,20 +23,26 @@ variable "prefix" {
   description = "Prefix to be used for resource names"
   type        = string
   default     = "materialize"
+  validation {
+    condition     = length(var.prefix) >= 3 && length(var.prefix) <= 16 && can(regex("^[a-z0-9-]+$", var.prefix))
+    error_message = "Prefix must be between 3-16 characters, lowercase alphanumeric and hyphens only."
+  }
 }
 
 variable "network_config" {
   description = "Network configuration for the AKS cluster"
   type = object({
-    vnet_address_space = string
-    subnet_cidr        = string
-    service_cidr       = string
+    vnet_address_space   = string
+    subnet_cidr          = string
+    postgres_subnet_cidr = string
+    service_cidr         = string
   })
   default = {
-    vnet_address_space = "10.0.0.0/16"
-    subnet_cidr        = "10.0.0.0/20"
-    service_cidr       = "10.1.0.0/16"
-    docker_bridge_cidr = "172.17.0.1/16"
+    vnet_address_space   = "10.0.0.0/16"
+    subnet_cidr          = "10.0.0.0/20"
+    postgres_subnet_cidr = "10.0.16.0/24"
+    service_cidr         = "10.1.0.0/16"
```

Comment:
I'd be more comfortable removing these variables and having them come from the examples